lib/assertions/exists.ts

```diff
@@ -3,15 +3,21 @@ export default function exists(options, message) {
     message = options;
     options = undefined;
   }
+  let label;
+  if (typeof this.target === 'string') {
+    label = this.target;
+  } else if (this.target instanceof Element) {
+    label = (this.target as Element).tagName;
+  }
```

Comment:
you might want to reuse the existing `targetDescription` property for this. but we should make sure to write tests for all the valid inputs (if they don't all exist yet) and check if the assertion messages are correct.

Reply:
I've moved some of this logic into `targetDescription`. Feel free to critique there!